gcore/rawdataset.h

```diff
@@ -63,11 +63,18 @@ class CPL_DLL RawDataset : public GDALPamDataset
          virtual ~RawDataset() = 0;
 
     bool GetRawBinaryLayout(GDALDataset::RawBinaryLayout&) override;
-    void ClearCachedConfigOption(void) {this->cachedCPLOneBigReadOption = std::make_pair(false, 0);}
+    void ClearCachedConfigOption(void) {CPLAtomicCompareAndExchange(&this->cachedCPLOneBigReadOption.all, this->cachedCPLOneBigReadOption.all, 0);}
```

Comment:
please move the implementation in gcore/rawdataset.cpp and add an explicit ``#include "cpl_atomic_ops.h"`` in it